qiskit_experiments/database_service/db_analysis_result.py

```diff
@@ -161,19 +153,34 @@ def save(self) -> None:
                 "Analysis result cannot be saved because no experiment service is available."
             )
             return
+        # Get DB fit data
+        value = self.value
+        result_data = {
+            "_value": json.loads(json.dumps(value, cls=self._json_encoder)),
+            "_extra": json.loads(json.dumps(self.extra, cls=self._json_encoder)),
+            "_source": self._source,
+        }
 
-        _result_data = json.loads(json.dumps(self._result_data, cls=self._json_encoder))
-        _result_data["_source"] = self._source
+        # Display compatible float values in in DB
+        if isinstance(value, (int, float, bool)):
+            result_data["value"] = float(value)
+        elif isinstance(value, FitVal):
+            if isinstance(value.value, (int, float, bool)):
+                result_data["value"] = float(value.value)
+            if isinstance(value.stderr, (int, float)):
+                result_data["variance"] = value.stderr ** 2
+            if isinstance(value.unit, str):
+                result_data["unit"] = value.unit
```

Comment:
If you take my suggestion in changing the `FitVal` json encoder, then this will become:
```suggestion
        elif isinstance(value, FitVal):
            result_data["value"] = result_data["_value"]
            result_data["value"]["variance"] = value.stderr ** 2
```

Reply:
I don't think this will work with DB display since it needs the float "value" "variance" "unit" fields as top level keys of the "result_data" (DB "fit") dict.

Reply:
Sorry, you are right. What I was going for was to make the json encoder for `FitVal` automatically produce that structure.

Reply:
```python
resullt_data.update(result_data["_value"])
```
should populate those keys up a level.

Reply:
Anyway, you can keep this section of code, but I would still recommend a more explicit encoding of `dataclass` objects as I suggested in json.py.